react-carousel 2.0: Introduce carousel plugins (#586)
brainhubeu/react-carousel: docs/migrationGuide.md

## Migration guide

> In the carousel v2 in order to make carousel more maintainable many props has been replaced with the plugins. In the following section you can find out hot to migrate the carousel v1 props into plugins.

Reviewer: ```suggestion
In the carousel v2 in order to make carousel more maintainable, many props have been replaced with the plugins. In the following section, you can find out how to migrate the carousel v1 props into plugins.
```

  <img src={imageThree} />
</Carousel>
```

####v2
```jsx
import { slidesToScrollPlugin } from '@brainhubeu/react-carousel';

<Carousel
  plugins={[
    {
      resolve: slidesToScrollPlugin,
      options: {
       numberOfSlides: 2,
      },
    },
  ]}   
>
  <img src={imageOne} />
  <img src={imageTwo} />
  <img src={imageThree} />
</Carousel>
```

### keepDirectionWhenDragging

####v1 
```jsx
<Carousel
  keepDirectionWhenDragging
>
  <img src={imageOne} />
  <img src={imageTwo} />
  <img src={imageThree} />
</Carousel>
```

####v2
```jsx
<Carousel
  plugins={[
    'fastSwipe',
  ]}
>
  <img src={imageOne} />
  <img src={imageTwo} />
  <img src={imageThree} />
</Carousel>
```

### autoplay

####v1 
```jsx
<Carousel
  autoPlay={2000}
  animationSpeed={1000}
>
  <img src={imageOne} />
  <img src={imageTwo} />
  <img src={imageThree} />
</Carousel>
```

####v2
```jsx
import { autoplayPlugin } from '@brainhubeu/react-carousel';

<Carousel
    plugins={[
    {
      resolve: autoplayPlugin,
      options: {
        interval: 2000,
      }
    },
  ]}   
  animationSpeed={1000}
>
  <img src={imageOne} />
  <img src={imageTwo} />
  <img src={imageThree} />
</Carousel>
```

### dots

> In v2 dots property has been removed. Please use [**dots component**](https://brainhubeu.github.io/react-carousel/docs/examples/dots) instead

Reviewer: ```suggestion
In v2 dots property has been removed. Please use [**dots component**](https://brainhubeu.github.io/react-carousel/docs/examples/dots) instead.
```
